mongoclient/db.go

```diff
@@ -38,62 +38,29 @@ type Config struct {
 	SecondaryPreferred bool
 }
 
-// Connect connects to MongoDB
-func Connect(config *Config, log *logrus.Entry) (*mongo.Client, error) {
-	opts := options.Client().
-		SetConnectTimeout(config.ConnTimeout).
-		SetReplicaSet(config.ReplSetName).
-		SetHosts(config.Servers)
-
-	if config.AppName != "" {
-		opts.SetAppName(config.AppName)
-	}
-
-	if config.SecondaryPreferred {
-		opts.SetReadPreference(readpref.SecondaryPreferred())
-	}
-
-	if config.TLS != nil && config.TLS.Enabled {
-		tlsLog := log.WithFields(logrus.Fields{
-			"cert_file": config.TLS.CertFile,
-			"key_file":  config.TLS.KeyFile,
-			"ca_files":  strings.Join(config.TLS.CAFiles, ","),
-		})
-
-		tlsLog.Debug("Using TLS config")
-		tlsConfig, err := config.TLS.TLSConfig()
-		if err != nil {
+func ConnectWithOptions(log logrus.FieldLogger, replSet string, servers []string, opts ...Option) (*mongo.Client, error) {
+	finalOpts := options.Client().
+		SetConnectTimeout(time.Minute). // use a default
```

Comment:
We want a config option for conn timeout right?

Reply:
This sets a sane default, I added an `Option` that will make this easy to override, but they can override it at any point (same with `replSet` and `servers` technically)